src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java

```diff
@@ -293,10 +293,7 @@ private Set<String> map(final User user, final TransportAddress caller) {
                 return Collections.emptySet();
             }
 
-            final Set<String> securityRoles = new HashSet<>();
-            synchronized (user.getSecurityRoles()) {
-                securityRoles.addAll(user.getSecurityRoles());
-            }
+            final Set<String> securityRoles = new HashSet<>(user.getSecurityRoles());
```

Comment:
The synchronization can be removed again because `user.getSecurityRoles()`  cannot be any longer concurrently modified as observed in https://github.com/opensearch-project/security/issues/4441